feat: Add support for `gh:` registry protocol for GitHub packages (#853)
vltpkg/vltpkg: www/docs/src/content/docs/cli/registries.mdx

the `registry` config option to a different value, such as a caching
proxy in your organization's network.

#### `gh:`

The `gh:` alias is a special alias that allows you to install packages

> from [GitHub Package Registry](https://github.com/features/packages).

Reviewer: That product page url does not seem to exist and is redirecting to a GitHub Actions product page. Let's point to the docs instead:

```suggestion
from [GitHub Package Registry](https://docs.github.com/en/packages).
```

For example, you can install a package from the GitHub package registry like
this:

```bash
$ vlt install gh:@octocat/hello-world@1.0.0
```

> This will fetch the package from `https://npm.pkg.github.com/` instead of
> requiring you to configure a scoped registry mapping in `.npmrc`.
> 

Reviewer: ```suggestion
This will fetch the package from `https://npm.pkg.github.com/` instead of
using the default registry.

```
